### PR TITLE
Use the new controller everywhere

### PR DESCRIPTION
**Pull request by richfitz**
This PR looks for as many instances of `con, keys` in argument lists that can be swapped out for `controller`; it's very noisy as a result

### CODE REVIEW

File: R/dependencies.R [hunk 1]
```diff
@@ -20,3 +23,48 @@ queue_dependencies <- function(con, keys, task_id, deferred_task_ids) {
...
+verify_dependencies_exist <- function(controller, depends_on) {
+  if (!is.null(depends_on)) {
+    dependencies_exist <- rrq_task_exists(depends_on, controller = controller)
+    if (!all(dependencies_exist)) {
+      missing <- names(dependencies_exist[!dependencies_exist])
+      error_msg <- ngettext(
+        length(missing),
+        "Failed to queue as dependency %s does not exist.",
+        "Failed to queue as dependencies %s do not exist.")
+      stop(sprintf(error_msg, paste0(missing, collapse = ", ")))
```
Comment (weshinsley) [inline R/dependencies.R:66]: to cli::abort at some point?

File: R/heartbeat.R [hunk 2]
```diff
@@ -18,7 +18,6 @@ cleanup_orphans <- function(controller, time) {
 
   con <- controller$con
```
Comment (weshinsley) [inline R/heartbeat.R:19]: con not needed below?
If not, it might be nicer to delete so it doesn't confuse with controller

File: R/dependencies.R [hunk 3]
```diff
@@ -20,3 +23,48 @@ queue_dependencies <- function(con, keys, task_id, deferred_task_ids) {
     con$pipeline(.commands = unlist(cmds, FALSE, FALSE))
   }
 }
+
+
+task_depends_down <- function(controller, task_ids) {
+  queue_id <- controller$queue_id
```
Comment (weshinsley) [inline R/dependencies.R:29]: I am confused in the PR about both 
controller$queue_id, and 
controller$keys$queue_id

File: R/rrq_controller.R [hunk 4]
```diff
@@ -1254,15 +1254,19 @@ rrq_controller <- R6::R6Class(
...
 }
 
 
-task_submit_n <- function(con, keys, store, task_ids, dat, key_complete, queue,
+task_submit_n <- function(controller, task_ids, dat, key_complete, queue,
                           separate_process, timeout, depends_on = NULL) {
+  con <- controller$con
+  keys <- controller$keys
+  store <- controller$store
+
```
Comment (weshinsley) [inline R/rrq_controller.R:1269]: I think you only need keys and con out of these three?

File: R/rrq_controller.R [hunk 5]
```diff
@@ -1357,16 +1361,20 @@ worker_naturalsort <- function(x) {
 }
 
 
-tasks_wait <- function(con, keys, store, task_ids, timeout, time_poll,
+tasks_wait <- function(controller, task_ids, timeout, time_poll,
                        progress, key_complete, error, follow,
                        call = NULL) {
+  con <- controller$con
+  keys <- controller$keys
+  store <- controller$store
```
Comment (weshinsley) [inline R/rrq_controller.R:1369]: store not strictly needed

**Review by weshinsley — APPROVED** [review]
OK - a few bits I'm not very clear on with how rrq works - but gone through everything with a few minor comments and it kind of makes sense...!

File: R/heartbeat.R (hunk 2 above)
Comment (richfitz) [reply R/heartbeat.R:19]: It's used further below the diff for `con$HMSET` and `con$SREM`

File: R/dependencies.R (hunk 3 above)
Comment (richfitz) [reply R/dependencies.R:29]: I think we should abandon keys$queue_id probably

File: R/rrq_controller.R (hunk 4 above)
Comment (richfitz) [reply R/rrq_controller.R:1269]: thanks, removed now

File: R/rrq_controller.R (hunk 5 above)
Comment (richfitz) [reply R/rrq_controller.R:1369]: thanks, removed now

File: R/dependencies.R (hunk 3 above)
Comment (richfitz) [reply R/dependencies.R:29]: turns out that we used keys$queue_id way more than controller$queue_id so dropped there instead

Comment (richfitz) [reply R/dependencies.R:29]: mrc-5249: tidying this properly might take a while

**Pull request merged**